Tutorial: How to set up Stripe reports (#8776)
PostHog/posthog.com: contents/tutorials/stripe-reports.md

---
title: How to set up Stripe reports
date: 2024-06-21
author:
 - ian-vanagas
tags:
 - data warehouse
---

> For your startup to become a real business, it needs to make money. Stripe is the most popular choice for helping you do this. Creating and analyzing reports for your Stripe data helps you understand how you are making money and how you can improve.
> 
> This tutorial shows you how to sync your Stripe data to PostHog and then create a report of insights like customer count, gross revenue, recurring revenue, revenue churn, and more.

Reviewer: Cutting out the first two sentences since it feels a bit like ELI5ing what stripe is
```suggestion
Creating and analyzing reports for your Stripe data helps you understand how you are making money and how you can improve. This tutorial shows you how to sync your Stripe data to PostHog and then create a report of insights like customer count, gross revenue, recurring revenue, revenue churn, and more.
```

## Linking Stripe data to PostHog

> To start, you need both a Stripe and PostHog account. Once have those, head to PostHog's [data warehouse tab](https://us.posthog.com/data-warehouse) and click **Link source**. Choose the Stripe option by clicking **Link**, and then enter your account ID and client secret, and press **Next**. Keep all tables selected and click **Import**. 

Reviewer: It would be better if they create a restricted key in stripe specifically for posthog, since the client secret grants us all sorts of access. I suspect we only need read access to a few things. Also, we should link to where they should create it: https://dashboard.stripe.com/apikeys/create

## Creating insights for your Stripe report

Now that your Stripe data is synced into PostHog, you can use it to create insights for your report. Each of these requires you to create a [new insight in the product analytics tab](https://us.posthog.com/project/insights/new).

### Customer count

To start, we create a trend of customer count over time. 

On the trends tab, change the aggregation to **Unique users**, and then click the data series, go to the **Data Warehouse** tab, hover over the `stripe_customer` table, and click **Select**. You might want to change the **Distinct ID field** from `id` to `email` as Stripe can give multiple `id` values to the same user email.

<ProductScreenshot
  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/customer_light_944920946a.png"
  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/customer_dark_e18fc2d09c.png"
  alt="Customer Count Insights"
  classes="rounded"
/>

This can also be done for any of the other Stripe data like charges, subscriptions, and invoices. You can also add filters based on property values like `created_at`, `email`, `status`, and more.

### Gross revenue

> Next, we can get our gross revenue by doing a similar process and selecting `stripe_charge`. For aggregation, we want **Property value sum** and then choose `amount`. We also want to filter out failed charges by clicking the filter button, selecting the `status` property, and making sure it doesn't equal `failed`.

Reviewer: I dont see the stripe_charge table in my warehouse 🤔 Maybe this is something to do with the fact I created a restricted key with read-only values for everything? I dont think so though.

Edit: I see it now. It was just taking time to sync all the data. We should make a note of this for users, since they're likely to run into this issue too

<img width="258" alt="Screenshot 2024-06-24 at 8 55 38 AM" src="https://github.com/PostHog/posthog.com/assets/7090054/0d42b61d-c0fd-4369-9319-79ac447f3f4f">

> Finally, to clean up the visualization, we can use formula mode to divide by 100 (the `amount` value is in cents) and click **Options** on the chart to add `$` as a prefix.

Reviewer: ```suggestion
Finally, to clean up the visualization, click **enable formula mode** to divide by 100 (the `amount` value is in cents) and click **Options** on the chart to add `$` as a prefix.
```

<ProductScreenshot
  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/revenue_light_bf87a357f2.png"
  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/revenue_dark_e75e42ffb3.png"
  alt="Gross Revenue Insights"
  classes="rounded"
/>

### Monthly recurring revenue

> There are many ways to calculate monthly recurring revenue, but the easiest and most common is multiplying the number of customers by the average revenue per customer per month.

Reviewer: It would be nice if we calculated MRR the [same way that stripe does](https://support.stripe.com/questions/calculating-monthly-recurring-revenue-(mrr)-in-billing#:~:text=Calculating%20Monthly%20Recurring%20Revenue%20(MRR,Billing%20%3A%20Stripe%3A%20Help%20%26%20Support&text=You%20can%20calculate%20Monthly%20Recurring,counts%20%24100%20towards%20your%20MRR.), since otherwise are numbers will differ whats on the stripe dashboard. 

It may be too complicated to do this though and require some HogQL. Up to you if you want to change it
